osquery/tables/system/windows/bios_info.cpp

```diff
@@ -0,0 +1,147 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <string>
+
+#include <osquery/tables.h>
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+
+#include <boost/regex.hpp>
+#include <boost/algorithm/string.hpp>
+
+namespace osquery {
+namespace tables {
+
+const auto kHPBiosSettingRegex = boost::regex("\\*([\\w ]*)");
+const std::vector<std::string> vHP = {
+    "hp", "hewlett-packard", "hewlett packard"};
+const std::vector<std::string> vLenovo = {"lenovo"};
+const std::vector<std::string> vDell = {"dell inc."};
+const std::map<std::string, std::pair<std::string, BSTR>> queryMap = {
+    {"hp",
+     {"select Name,Value from HP_BiosSetting", L"root\\hp\\instrumentedBIOS"}},
+    {"lenovo", {"select CurrentSetting from Lenovo_BiosSetting", L"root\\wmi"}},
+    {"dell",
+     {"select AttributeName,CurrentValue,PossibleValues, "
+      "PossibleValuesDescription from DCIM_BIOSEnumeration",
+      L"root\\dcim\\sysman"}}};
+
+std::string getManufacturer(std::string manufacturer) {
+  transform(manufacturer.begin(),
+            manufacturer.end(),
+            manufacturer.begin(),
+            ::tolower);
+
+  if (std::find(vHP.begin(), vHP.end(), manufacturer) != vHP.end()) {
+    manufacturer = "hp";
+  } else if (std::find(vLenovo.begin(), vLenovo.end(), manufacturer) !=
+             vLenovo.end()) {
+    manufacturer = "lenovo";
+  } else if (std::find(vDell.begin(), vDell.end(), manufacturer) !=
+             vDell.end()) {
+    manufacturer = "dell";
+  }
+
+  return manufacturer;
+}
+
+QueryData genBiosInfo(QueryContext& context) {
+  QueryData results;
+  std::string manufacturer;
+
+  WmiRequest wmiComputerSystemReq(
+      "select Manufacturer from Win32_ComputerSystem");
+  std::vector<WmiResultItem>& wmiComputerSystemResults =
+      wmiComputerSystemReq.results();
+
+  if (!wmiComputerSystemResults.empty()) {
+    wmiComputerSystemResults[0].GetString("Manufacturer", manufacturer);
+    manufacturer = getManufacturer(manufacturer);
+  } else {
+    manufacturer = "N/A";
+  }
+
+  if (queryMap.find(manufacturer) != queryMap.end()) {
```

Comment:
Same deal here. Flip your logic and return early if the manufacturer isn’t in the list you have and reduce your scope.